csharp/src/SeedLang/Interpreter/RegisterAllocator.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace SeedLang.Interpreter {
+  // The allocator class to allocate register slots for local and temporary variables.
+  internal class RegisterAllocator {
+    // The maximum allocated registers count. This number is never decreased even when a register is
+    // deallocated.
+    public uint MaxRegisterCount = 0;
```

Comment:
What's the use of `MaxRegisterCount`? If the filed `_registerCount` is pointed to the current boundary of the allocation, what's the reason that we need to get an ever-reached maximum number?

Seems there is no other code using `MaxRegisterCount` in this PR. Is it useful in the future somehow?

Reply:
The compiler will copy MaxRegisterCount into Chunk after compilation, which will be used as the allocation size of the registers in the stack during execution. Added code to handle the copy logic. The register allocation logic during execution will be added later.